draft-ietf-acme-acme.md

```diff
@@ -955,18 +948,17 @@ key but not the corresponding registration URI to recover the registration URI.
 
 If the server wishes to present the client with terms under which the ACME
 service is to be used, it MUST indicate the URI where such terms can be accessed
-in a Link header with link relation "terms-of-service".  As noted above, the
-client may indicate its agreement with these terms by updating its registration
-to include the "agreement" field, with the terms URI as its value.  When these
-terms change in a way that requires an agreement update, the server MUST
-use a different URI in the Link header.
+in the "terms-of-service" subfield of the "meta" field in the directory object,
+and the server MUST reject new-registration requests that do not contain
+"terms-of-service-agreed": true. Note that the "terms-of-service-agreed" field
+is not echoed in the resulting registration object, because it is a one-time
+field.
```

Comment:
I'm not loving "because it is a one-time field".  Basically, you're saying there's no need to reflect this on the registration object, because it's implicitly true given that the server rejects new-reg when it's false.

I don't think that's actually true, though, if the server can make breaking changes to ToS and clients need to re-agree.  In that case, there's no way to tell from the registration object whether you've already re-agreed or not; you just have to try something and get the error.  And in any case, explicit state > implicit state.

Net/net, I would prefer if we could just have it be a field on the object; the server's going to have to manage it either way.